src/app/state/index.ts

```diff
@@ -0,0 +1,36 @@
+import {
+  // ActionReducer,
+  ActionReducerMap,
+  createFeatureSelector,
+  createSelector,
+  MetaReducer
+} from '@ngrx/store';
+import { environment } from '../../environments/environment';
+import { Course, User } from '../models/data-model';
+import { coursesReducer } from './courses/courses.reducer';
+import { userReducer } from './user/user.reducer';
+
+
+export interface State {
+  courses: CourseState;
+  user: UserState;
+}
+
+export const reducers: ActionReducerMap<State> = {
+  courses: coursesReducer,
+  user: userReducer,
+};
+
+export interface CourseState {
+  courses: Course[];
```

Comment:
The difference between `courses`, `allCourses` and `currentCourses` is not clear. Please consider renaming to make code easier to understand